src/tools/components/SequenceSearchLoader.tsx

```diff
@@ -0,0 +1,245 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  ChangeEvent,
+  ClipboardEvent,
+  forwardRef,
+  useImperativeHandle,
+  useCallback,
+} from 'react';
+import queryString from 'query-string';
+import { SearchInput } from 'franklin-sites';
+import { useDispatch } from 'react-redux';
+import { v1 } from 'uuid';
+
+import useDataApi from '../../shared/hooks/useDataApi';
+
+import { addMessage } from '../../messages/state/messagesActions';
+
+import uniProtKBApiUrls from '../../uniprotkb/config/apiUrls';
+import uniParcApiUrls from '../../uniparc/config/apiUrls';
+
+import entryToFASTAWithHeaders from '../../uniprotkb/adapters/entryToFASTAWithHeaders';
+import { uniProtKBAccessionRegEx } from '../../uniprotkb/utils';
+import fetchData from '../../shared/utils/fetchData';
+
+import {
+  MessageFormat,
+  MessageLevel,
+} from '../../messages/types/messagesTypes';
+import { APISequenceData } from '../blast/types/apiSequenceData';
+import { EntryType } from '../../uniprotkb/adapters/uniProtkbConverter';
+
+const getURLForAccessionOrID = (input: string) => {
+  const cleanedInput = input.trim().toUpperCase();
```

Comment:
'there is nothing to prevent the function to be called without a valid string value', well, that is definitely why we have TypeScript 🤔 I don't really see why we'd need to add an extra check here

Reply:
So my understanding is that TypeScript is a compile-time compiler, meaning it won't be available in run-time. This is something many people hate about TypeScript, but it has been communicated from the development team that TSC will stay, forever, a compile-time compiler. This is because .ts files should be compiler to .js files and at that point everything is JavaScript.

This is why for example you can't (easily) do type checking on custom classes via TSC, simply because the type information is not available after compile, it's all plain JavaScript.

The problem that may occur here is that when we receive data from API or user for example, we might have assumed that there will be an array there full of strings and we have reflected that in our type definitions. However, the API might output an array of arrays full of strings either due to an error or a change.

TSC will not and have no way of checking this, because when the code is running it is JS. In simple words, `input: string` is all fine during the coding and compile-time, but once the code is compiled it will be completely stripped of any type-checks and guards.

Example here: [https://www.typescriptlang.org/play/#code/GYVwdgxgLglg9mABAUwgCzgCgLbIM54CGA5sgFyJ5QBOMYxAlIgN4CwAUIotclCNUlwESyANwcAvkA](https://www.typescriptlang.org/play/#code/GYVwdgxgLglg9mABAUwgCzgCgLbIM54CGA5sgFyJ5QBOMYxAlIgN4CwAUIotclCNUlwESyANwcAvkA)

Reply:
While I understand your point that TypeScript annotations are only validated at compile time:
 - This specific function is never used with data coming from an API. It's used twice with data coming from user input in a *text* field (1 is from typing, the other from pasting)
 - I don't understand why we would need to check _here_ but not in the literally hundreds of others places in the codebase where we do assume that the data we get, some from APIs as you say, is actually of the schema/TypeScript definition we say it is.